Feature/hide empty tabs

nnn opened the pull request:
Closes #1539

brylie commented on apis/collection/helpers.js:
```diff
@@ -237,4 +239,43 @@ Apis.helpers({
 
     return documentation;
   },
+  backlogIsNotEmpty () {
```
This function name is misleading. In effect, the function is doing two things:

- checking if there are any backlog items
- checking if user can manage

So, a more general name would suffice here, such as `userShouldSeeBacklogTab`.

brylie commented on apis/collection/helpers.js:
```diff
@@ -237,4 +239,43 @@ Apis.helpers({
+    }
+    return false;
+  },
+  apiDocsIsNotEmpty () {
```
This function name is misleading. In effect, the function is doing two things:

- checking if there are any ApiDocs items
- checking if user can manage

So, a more general name would suffice here, such as `userShouldSeeApiDocab`.

brylie commented on apis/collection/helpers.js:
```diff
@@ -237,4 +239,43 @@ Apis.helpers({
+    }
+    return false;
+  },
+  apiMetadataIsNotEmpty () {
```
This function name is misleading. In effect, the function is doing two things:

- checking if there are any API Metadata
- checking if user can manage

So, a more general name would suffice here, such as `userShouldSeeApiMetadataTab`.

brylie commented:
I just realized that the functions are attached to the collection, as collection helpers. Why do we use collection helpers instead of template helpers? What is the main purpose of the functions?

brylie commented:
@nnn is this ready?

nnn commented:
Removed ApiDocs tab related code.
Anonymous is not able to see documentation, even if it exist.
Also getting weird console errors: 

```
I20170330-12:22:26.139(3)? Exception in queued task: TypeError: this.cursor._getCollectionName is not a function
I20170330-12:22:26.139(3)?     at Publication._getCollectionName (packages/reywood_publish-composite.js:140:63)
I20170330-12:22:26.139(3)?     at Publication.publish (packages/reywood_publish-composite.js:81:31)
I20170330-12:22:26.140(3)?     at Publication.createChildPublication (packages/reywood_publish-composite.js:147:13)
I20170330-12:22:26.140(3)?     at Array.forEach (native)
I20170330-12:22:26.140(3)?     at Function._.each._.forEach (packages/underscore.js:139:11)
I20170330-12:22:26.140(3)?     at Publication._publishChildrenOf (packages/reywood_publish-composite.js:144:7)
I20170330-12:22:26.140(3)?     at Object.added (packages/reywood_publish-composite.js:96:22)
I20170330-12:22:26.140(3)?     at [object Object].observeChangesCallbacks.added (packages/minimongo/observe.js:153:28)
I20170330-12:22:26.140(3)?     at self.applyChange.added (packages/minimongo/observe.js:53:44)
I20170330-12:22:26.141(3)?     at packages/mongo/observe_multiplex.js:182:30
I20170330-12:22:26.141(3)?     at Array.forEach (native)
I20170330-12:22:26.141(3)?     at Function._.each._.forEach (packages/underscore.js:139:11)
I20170330-12:22:26.142(3)?     at Object.task (packages/mongo/observe_multiplex.js:176:9)
I20170330-12:22:26.142(3)?     at [object Object]._.extend._run (packages/meteor.js:807:18)
I20170330-12:22:26.142(3)?     at packages/meteor.js:785:14
```
```

I20170330-11:59:10.777(3)? Exception from sub organizationApiLinksByOrganizationSlug id jPSHt5J72a2KZPCh9 TypeError: Cannot read property '_id' of undefined
I20170330-11:59:10.778(3)?     at [object Object].<anonymous> (organization_apis/collection/server/publications.js:51:26)
I20170330-11:59:10.779(3)?     at packages/matb33_collection-hooks.js:307:21
I20170330-11:59:10.780(3)?     at [object Object]._.extend.withValue (packages/meteor.js:1122:17)
I20170330-11:59:10.781(3)?     at [object Object]._handler (packages/matb33_collection-hooks.js:306:28)
I20170330-11:59:10.782(3)?     at maybeAuditArgumentChecks (packages/ddp-server/livedata_server.js:1737:12)
I20170330-11:59:10.782(3)?     at [object Object]._.extend._runHandler (packages/ddp-server/livedata_server.js:1035:17)
I20170330-11:59:10.783(3)?     at [object Object]._.extend._startSubscription (packages/ddp-server/livedata_server.js:853:9)
I20170330-11:59:10.783(3)?     at [object Object]._.extend.protocol_handlers.sub (packages/ddp-server/livedata_server.js:625:12)
I20170330-11:59:10.783(3)?     at packages/ddp-server/livedata_server.js:559:43
```

nnn commented:
@brylie yep.
